### PR TITLE
nicer exceptions for model registry

### PR DESCRIPTION
**Pull request by pkasprzyk**

### CODE REVIEW

File: neptune/new/exceptions.py [hunk 1]
```diff
@@ -770,18 +800,41 @@ def __init__(self):
...
+                extra_info=self.extra_info.format(**STYLES),
+                **STYLES,
+            )
+        )
+
+
+class NeptuneCannotChangeStageManually(NeptuneProtectedPathException):
+    extra_info = """
+If you want to change the stage of the model version,
+use the {python}.change_stage(){end} function."""
```
Comment (pkasprzyk) [inline neptune/new/exceptions.py:837]: image

File: neptune/new/exceptions.py [hunk 2]
```diff
@@ -618,6 +624,30 @@ def __init__(
         )
 
 
+class NeptuneMissingRequiredInitParameter(NeptuneWrongInitParametersException):
```
Comment (pkasprzyk) [inline neptune/new/exceptions.py:627]: example:
image

File: neptune/new/exceptions.py [hunk 3]
```diff
@@ -392,15 +392,21 @@ def __init__(self, label: str):
 
 class InactiveModelException(InactiveContainerException):
     resume_info = """
-    TODO: NPT-11349"""
+    - Resume the {container_type} to continue logging to it.
+    - Don't invoke `stop()` on a {container_type} that you want to access. If you want to stop monitoring only,
+    you can resume a {container_type} in read-only mode:
+    https://docs.neptune.ai/you-should-know/connection-modes#read-only"""
```
Comment (pkasprzyk) [inline neptune/new/exceptions.py:398]: simple placeholder (same for Model and ModelVersion):
image

File: neptune/new/metadata_containers/model.py [hunk 4]
```diff
@@ -43,7 +43,13 @@ def _label(self) -> str:
         return self._sys_id
 
     def fetch_model_versions_table(self) -> Table:
-        """TODO: NPT-11349"""
+        """Retrieve all model versions of the given model.
```
Comment (pkasprzyk) [inline neptune/new/metadata_containers/model.py:46]: @Herudaio nb. no filtering on model versions for day 1. Is this OK?

File: neptune/new/exceptions.py (hunk 1 above)
Comment (Herudaio) [reply neptune/new/exceptions.py:837]: I would provide a full example:
`If you want to change the stage of the model version,
use the {python}.change_stage(){end} function:
    {python}model_version.change_stage("staging"){end}`

File: neptune/new/exceptions.py (hunk 3 above)
Comment (Herudaio) [reply neptune/new/exceptions.py:398]: 2. Content for resuming:

```
    - Resume the model to continue logging to it:
    https://docs.neptune.ai/api-reference/neptune#.init_model
    ```

```
    - Resume the model version to continue logging to it:
    https://docs.neptune.ai/api-reference/neptune#.init_model_version
    ```

File: neptune/new/exceptions.py (hunk 1 above)
Comment (pkasprzyk) [reply neptune/new/exceptions.py]: update, image

File: neptune/new/exceptions.py (hunk 3 above)
Comment (Herudaio) [reply neptune/new/exceptions.py:402]: 1. Why `model_version` not "model version"?
2. In the second part there is still {container_type}

File: neptune/new/exceptions.py (hunk 2 above)
Comment (Herudaio) [reply neptune/new/exceptions.py:632]:
```{python}neptune.{called_function}(){end} is missing {parameter_name} parameter. 

If you want to create a new object using {called_function}, {parameter_name} is required:
https://docs.neptune.ai/api-reference/neptune#.init_model_version

Need help? [...]
```

Links:
• https://docs.neptune.ai/api-reference/neptune#.init_model_version
• https://docs.neptune.ai/api-reference/neptune#.init_model

Comment (pkasprzyk) [reply neptune/new/exceptions.py:632]: fixed

File: neptune/new/exceptions.py (hunk 3 above)
Comment (pkasprzyk) [reply neptune/new/exceptions.py:402]: Ad 2. Fixed, sorry
Ad 1. That's the value of our enum -- we use it in places where space is not really a safe option (e.g. filesystem paths). We're using variables to limit copy&paste, as we already have a lot of text in `exceptions.py`. Could you perhaps live with this underscore? :)

File: neptune/new/metadata_containers/model.py (hunk 4 above)
Comment (pkasprzyk) [reply neptune/new/metadata_containers/model.py:46]: discussed offline, resolving

File: neptune/new/exceptions.py (hunk 3 above)
Comment (Herudaio) [reply neptune/new/exceptions.py:402]: I can and I should, I will always ask "maybe?" to push toward user :)

**Review by Herudaio — APPROVED** [review]

**Review by Raalsky — APPROVED** [review]

**Pull request merged**